### PR TITLE
fix for eslint rule protecting against importing packages that aren't in the package.json

### PR DESCRIPTION
**Pull request by alshdavid**
# ↪️ Pull Request

This PR corrects the misbehaving `eslint-plugin-import/no-extraneous-dependencies` rule, adding protection against using imports that are not defined in a package's package.json 

_Note:_ 
This currently excludes type imports like `import type { Foo } from 'foo'` due to the current version of [eslint-plugin-import](https://www.runpkg.com/?eslint-plugin-import@2.26.0/lib/rules/no-extraneous-dependencies.js#239) not including [the change which adds support for type imports](https://github.com/import-js/eslint-plugin-import/blob/main/src/rules/no-extraneous-dependencies.js#L272).

I am not sure when they will merge this, the last update was almost a year ago.

## 💻 Examples

```typescript
import('foo')

// Will fail to lint unless the closest "package.json" has "foo"
// in either the dependencies or devDependencies
```

## 🚨 Test instructions

As this is a linting rule, to test it you can try importing a package that isn't in that package.json to see if it fails

## ✔️ PR Todo

- [ ] Added/updated unit tests for this change
- [ ] Filled out test instructions (In case there aren't any unit tests)
- [ ] Included links to related issues/PRs

### CODE REVIEW

**Comment by parcel-benchmark** [comment]
## Benchmark Results
<details><summary>Kitchen Sink ✅</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | 1.45s | +28.00ms |
| Cached | 319.00ms | +13.00ms |

#### Cold Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/legacy/index.html | 826.00b | -3.00b 🚀 | 392.00ms | -5.00ms |

#### Cached Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/legacy/parcel.7cdb0fad.webp | 102.94kb | +0.00b | 233.00ms | +19.00ms ⚠️ |
| dist/legacy/parcel.7cdb0fad.webp | 102.94kb | +0.00b | 234.00ms | +20.00ms ⚠️ |
| dist/modern/parcel.7cdb0fad.webp | 102.94kb | +0.00b | 233.00ms | +19.00ms ⚠️ |
| dist/legacy/index.html | 826.00b | -3.00b 🚀 | 417.00ms | +10.00ms |
| dist/legacy/index.b8ae99ba.css | 94.00b | +0.00b | 244.00ms | +22.00ms ⚠️ |
| dist/modern/index.31cedca9.css | 94.00b | +0.00b | 243.00ms | +22.00ms ⚠️ |

</p></details>

<details><summary>React HackerNews ✅</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | 6.81s | +75.00ms |
| Cached | 410.00ms | -7.00ms |

#### Cold Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/PermalinkedComment.60e78a07.js | 4.18kb | +0.00b | 400.00ms | -27.00ms 🚀 |
| dist/UserProfile.c18819ee.js | 1.57kb | +0.00b | 400.00ms | -27.00ms 🚀 |
| dist/NotFound.cfeedbab.js | 427.00b | +0.00b | 400.00ms | -27.00ms 🚀 |

#### Cached Bundles

*No bundle changes detected.*

</p></details>

<details><summary>AtlasKit Editor ✅</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | 59.11s | -1.27s |
| Cached | 1.99s | +94.00ms |

#### Cold Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/media-viewer.bd165005.js | 542.15kb | +0.00b | 6.95s | -816.00ms 🚀 |
| dist/ConfigPanelFieldsLoader.f06a6b36.js | 312.08kb | +0.00b | 6.95s | -815.00ms 🚀 |
| dist/card.501ecffa.js | 143.52kb | +0.00b | 6.95s | -815.00ms 🚀 |
| dist/ConfigPanelFieldsLoader.e1ae433f.js | 83.45kb | +0.00b | 6.95s | -815.00ms 🚀 |
| dist/ElementBrowser.3bcad544.js | 65.85kb | +0.00b | 6.95s | -814.00ms 🚀 |
| dist/archive.503fa405.js | 61.48kb | +0.00b | 9.67s | +1.91s ⚠️ |
| dist/esm.945b66be.js | 60.94kb | +0.00b | 6.95s | -815.00ms 🚀 |
| dist/component-lazy.60375b05.js | 60.45kb | +0.00b | 5.14s | +347.00ms ⚠️ |
| dist/ConfigPanelFieldsLoader.ef739802.js | 16.14kb | +0.00b | 6.95s | -814.00ms 🚀 |
| dist/ui.2de0ef21.js | 14.88kb | +0.00b | 6.95s | -815.00ms 🚀 |
| dist/ConfigPanelFieldsLoader.c68d84ab.js | 14.25kb | +0.00b | 6.95s | -814.00ms 🚀 |
| dist/pdfRenderer.187ba54d.js | 12.21kb | +0.00b | 6.95s | -815.00ms 🚀 |
| dist/mobile-upload.136dd5cb.js | 8.08kb | +0.00b | 6.95s | -814.00ms 🚀 |
| dist/mobile-upload.0bdb676c.js | 8.08kb | +0.00b | 6.95s | -815.00ms 🚀 |
| dist/index.b16227d6.css | 4.08kb | -2.00b 🚀 | 9.79s | -154.00ms |
| dist/media-viewer-analytics-error-boundary.e6109a6a.js | 3.46kb | +0.00b | 9.67s | +1.91s ⚠️ |
| dist/ru.896915b9.js | 2.94kb | +0.00b | 6.95s | +1.66s ⚠️ |
| dist/uk.48c97550.js | 2.89kb | +0.00b | 6.95s | -814.00ms 🚀 |
| dist/codeViewerRenderer.915ef6b3.js | 2.84kb | +0.00b | 9.67s | +1.91s ⚠️ |
| dist/th.31044730.js | 2.73kb | +0.00b | 6.95s | -814.00ms 🚀 |
| dist/vi.d8dcb67a.js | 2.22kb | +0.00b | 6.95s | -814.00ms 🚀 |
| dist/tr.46f26598.js | 2.16kb | +0.00b | 6.95s | -814.00ms 🚀 |
| dist/sv.13d93533.js | 2.10kb | +0.00b | 6.95s | -814.00ms 🚀 |
| dist/zh_TW.afaf6222.js | 1.98kb | +0.00b | 6.95s | -814.00ms 🚀 |
| dist/zh.fcdc32bb.js | 1.96kb | +0.00b | 6.95s | -814.00ms 🚀 |
| dist/workerHasher.ef49a7fc.js | 1.72kb | +0.00b | 6.95s | -814.00ms 🚀 |
| dist/workerHasher.9d5fe27b.js | 1.72kb | +0.00b | 6.95s | -815.00ms 🚀 |
| dist/heading5.023a8f1f.js | 1.36kb | +0.00b | 5.14s | +348.00ms ⚠️ |
| dist/sk.101f1705.js | 786.00b | +0.00b | 6.95s | +1.66s ⚠️ |
| dist/simpleHasher.f1f58b0a.js | 687.00b | +0.00b | 6.95s | -814.00ms 🚀 |
| dist/simpleHasher.09f4d713.js | 687.00b | +0.00b | 6.95s | -815.00ms 🚀 |
| dist/ro.a6eff34a.js | 612.00b | +0.00b | 6.95s | +1.66s ⚠️ |
| dist/index.html | 240.00b | +0.00b | 9.83s | +5.36s ⚠️ |

#### Cached Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/media-viewer.bd165005.js | 542.15kb | +0.00b | 7.01s | -1.11s 🚀 |
| dist/index.b16227d6.css | 4.08kb | -2.00b 🚀 | 9.78s | +309.00ms |
| dist/heading4.05995ed9.js | 1.25kb | +0.00b | 5.07s | +440.00ms ⚠️ |

</p></details>

<details><summary>Three.js ✅</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | 4.76s | -16.00ms |
| Cached | 265.00ms | +1.00ms |

#### Cold Bundles

*No bundle changes detected.*

#### Cached Bundles

*No bundle changes detected.*

</p></details>

**[Click here to view a detailed benchmark overview.](https://benchmark.parceljs.org/benchmark/360266994865406532)**

File: packages/utils/node-resolver-core/package.json [hunk 1]
```diff
@@ -47,5 +49,8 @@
     "url": "^0.11.0",
     "util": "^0.12.3",
     "vm-browserify": "^1.1.2"
+  },
+  "peerDependencies": {
+    "@parcel/core": "2.8.0"
```
Comment (mischnic) [inline packages/utils/node-resolver-core/package.json]: Looks like this is only used in the tests, so the peerdependency isn't needed?

File: packages/core/package-manager/package.json [hunk 2]
```diff
@@ -46,7 +46,8 @@
     "@parcel/types": "2.8.0",
     "@parcel/utils": "2.8.0",
     "@parcel/workers": "2.8.0",
-    "semver": "^5.7.1"
+    "semver": "^5.7.1",
+    "sinon": "^7.3.1"
```
Comment (mischnic) [inline packages/core/package-manager/package.json]: sinon should be a devdep?

File: packages/core/logger/package.json [hunk 3]
```diff
@@ -21,6 +21,7 @@
   },
   "dependencies": {
     "@parcel/diagnostic": "2.8.0",
-    "@parcel/events": "2.8.0"
+    "@parcel/events": "2.8.0",
+    "sinon": "^7.3.1"
```
Comment (mischnic) [inline packages/core/logger/package.json:27]: sinon should be a devdep

File: packages/core/graph/package.json [hunk 4]
```diff
@@ -21,6 +21,7 @@
   },
   "dependencies": {
     "@parcel/utils": "2.8.0",
-    "nullthrows": "^1.1.1"
+    "nullthrows": "^1.1.1",
+    "sinon": "^7.3.1"
```
Comment (mischnic) [inline packages/core/graph/package.json:26]: sinon should be a devdep

File: packages/core/core/package.json [hunk 5]
```diff
@@ -47,9 +48,11 @@
     "json5": "^2.2.0",
     "msgpackr": "^1.5.4",
     "nullthrows": "^1.1.1",
-    "semver": "^5.7.1"
+    "semver": "^5.7.1",
+    "sinon": "^7.3.1"
```
Comment (mischnic) [inline packages/core/core/package.json]: sinon should be a devdep

File: packages/core/graph/package.json (hunk 4 above)
Comment (alshdavid) [reply packages/core/graph/package.json:26]: Updated

File: packages/core/graph/src/shared-buffer.js [hunk 6]
```diff
@@ -1,5 +1,4 @@
 // @flow
-/* global MessageChannel:readonly */
```
Comment (alshdavid) [inline packages/core/graph/src/shared-buffer.js:2]: This is likely due to the update to eslint

Screenshot from 2023-01-11 15-23-12

File: packages/core/logger/package.json (hunk 3 above)
Comment (alshdavid) [reply packages/core/logger/package.json:27]: Thanks, updated

**Comment by mischnic** [comment]
The tests are failing now, and there's a merge conflict

File: packages/core/core/package.json [hunk 7]
```diff
@@ -26,6 +26,7 @@
   "dependencies": {
     "@mischnic/json-sourcemap": "^0.1.0",
     "@parcel/cache": "2.8.2",
+    "@parcel/config-default": "2.8.2",
```
Comment (mischnic) [inline packages/core/core/package.json]: This should also be a dev dependency

**Comment by mischnic** [comment]
> This currently excludes type imports like import type { Foo } from 'foo' due to the current version of [eslint-plugin-import](https://www.runpkg.com/?eslint-plugin-import@2.26.0/lib/rules/no-extraneous-dependencies.js#239) not including [the change which adds support for type imports](https://github.com/import-js/eslint-plugin-import/blob/main/src/rules/no-extraneous-dependencies.js?rgh-link-date=2022-11-14T06%3A17%3A23Z#L272).
> 
> I am not sure when they will merge this, the last update was almost a year ago.

BTW, that was released in the meantime with 2.27

**Pull request closed without merging**